### PR TITLE
Re-format all markdown with prettier and make it an optional check

### PR DESCRIPTION
**Pull request by wlach**
For now we won't make it required, to reduce friction.

### CODE REVIEW

File: src/cookbooks/public_data.md [hunk 1]
```diff
@@ -32,54 +32,56 @@ For example:
...
-                "incremental": true,
-                "incremental_export": false,
-                "review_link": "https://bugzilla.mozilla.org/show_bug.cgi?id=1414839",
-                "files_uri": "https://public-data.telemetry.mozilla.org/api/v1/tables/telemetry_derived/ssl_ratios/v1/files",
-                "last_updated": "https://public-data.telemetry.mozilla.org/api/v1/tables/telemetry_derived/ssl_ratios/v1/last_updated"
-            }
-        },
-        // [...]
+  "telemetry_derived": {
+    // dataset name
```
Comment (mreid-moz) [inline src/cookbooks/public_data.md:36]: This change seems to make the "comments" less readable.

Comment (mreid-moz) [inline src/cookbooks/public_data.md:36]:
```suggestion
    // ^ dataset name
```

File: src/cookbooks/public_data.md [hunk 2]
```diff
@@ -32,54 +32,56 @@ For example:
...
+    "deviations": {
+      // table name
```
Comment (mreid-moz) [inline src/cookbooks/public_data.md:38]:
```suggestion
      // ^ table name
```

File: src/cookbooks/public_data.md [hunk 3]
```diff
@@ -32,54 +32,56 @@ For example:
...
+      "v1": {
+        // table version
```
Comment (mreid-moz) [inline src/cookbooks/public_data.md:40]:
```suggestion
        // ^ table version
```

File: src/cookbooks/publishing_datasets.md [hunk 4]
```diff
@@ -18,30 +18,31 @@ owners:
   - example@mozilla.com
 labels:
   application: firefox
-  incremental: true     # incremental queries add data to existing tables
-  schedule: daily       # scheduled in Airflow to run daily
+  incremental: true # incremental queries add data to existing tables
```
Comment (mreid-moz) [inline src/cookbooks/publishing_datasets.md:21]: Ditto - the comments were aligned vertically before, making it easier to read them.

**Review by mreid-moz — APPROVED** [review]
This looks good to me - ~there are a few places where (imo) the changes make the markdown slightly harder to read, such as whitespace around comments in code blocks and to a lesser extent table formatting which sometimes results in very long lines~. After writing this I read [`prettier`s philosophy page](https://prettier.io/docs/en/why-prettier.html) and agree that these minor issues are worth it for consistent style.

This is a nice improvement! Thanks!

File: src/cookbooks/publishing_datasets.md (hunk 4 above)
Comment (wlach) [reply src/cookbooks/publishing_datasets.md:21]: Looks like someone else had the same opinion: prettier/prettier#8690

I don't think this is really worth worrying about. It's still quite readable IMO.

**Pull request merged**